File: app/src/main/java/com/mohsenoid/rickandmorty/data/db/dto/DbOriginModel.java
package com.mohsenoid.rickandmorty.data.db.dto;

public class DbOriginModel {

    private String name;
    private String url;

    public DbOriginModel(String name, String url) {
        this.name = name;
        this.url = url;
    }

    public String getName() {
        return name;
    }

    public String getUrl() {
        return url;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        DbOriginModel that = (DbOriginModel) o;
        return name.equals(that.name) &&
                url.equals(that.url);
    }
}

File: app/src/main/kotlin/com/mohsenoid/rickandmorty/data/db/dto/DbOriginModel.kt
package com.mohsenoid.rickandmorty.data.db.dto

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import kotlinx.serialization.Serializable

@Serializable
@Entity(tableName = "origins")
data class DbOriginModel(

    @PrimaryKey
    @ColumnInfo(name = "name")
    val name: String,

    @ColumnInfo(name = "url")
    val url: String
)
